### PR TITLE
Create reproducible .tar.gz release tarball + documentation fixes

### LINKED ISSUE #11: Reproducible tar.gz release tarball
The generated `.tar.gz` files are not reproducible.
This is caused by embedding a timestamp by python.

See https://github.com/python/cpython/blob/802726acf6048338394a6a4750835c2cdd6a947b/Lib/tarfile.py#L419-L420

### PR DESCRIPTION
**Pull request by madebr**
- Documentation fixes
- Make .tar.gz reproducible
- Add `.editorconfig` for autoconfiguring IDE's (https://editorconfig.org/)

Fixes #11

### CODE REVIEW

File: Changelog [hunk 1]
```diff
@@ -12,8 +12,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 (nothing)
 
 ### Changed
-
-(nothing)
+- Make .tar.gz release tarballs reproducible by zero'ing the time stamp
```
Comment (kulp) [inline Changelog:15]: Nit: maybe insert a blank line before this item for consistency within this file (although I think perhaps soon we should remove some of the extra blank lines, so I do not care very much).

Comment (madebr) [reply Changelog:15]: Indeed a blank looks better.
I'll leave it for the next comitter :smile:

File: create_source_tarball.py [hunk 2]
```diff
@@ -28,3 +28,11 @@ def packaged_sources(distfiles_yml: Path):
             name = Path(tar_filebasename) / file.relative_to(WORKPATH)
             print("Adding {}...".format(name))
             tar.add(name=file, arcname=str(name))
+
+
+# Zero the embedded timestamp in the gzip'ed tarball
+targz = Path("naive-tsearch-{}.tar.gz".format(version))
+if targz.is_file():
+    with targz.open("r+b") as f:
```
[the remark this answers is not in the document]
Comment (madebr) [reply create_source_tarball.py:36]: The first tests for existence of the file.
But it should always be there.
If it does not, then there is something seriously wrong with the system.
I'll remove the first indent.

File: create_source_tarball.py [hunk 3]
```diff
@@ -28,3 +28,11 @@ def packaged_sources(distfiles_yml: Path):
...
+        f.seek(4, 0)
+        f.write(b"\x00\x00\x00\x00")
```
Comment (madebr) [inline create_source_tarball.py:38]:
```suggestion
with open("naive-tsearch-{}.tar.gz".format(version), "r+b") as f:
    f.seek(4, 0)
    f.write(b"\x00\x00\x00\x00")
```

File: create_source_tarball.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kulp) [reply create_source_tarball.py]: If I remember correctly, the `with` will throw an exception if the file does not exist, which is probably good.

**Comment by kulp** [comment]
Thanks !

**Pull request merged**